controllers/argocd/argocd_controller.go

```diff
@@ -104,90 +106,109 @@ func (r *ReconcileArgoCD) Reconcile(ctx context.Context, request ctrl.Request) (
 		return reconcile.Result{}, err
 	}
 
-	newPhase := argocd.Status.Phase
-	// If we discover a new Argo CD instance in a previously un-seen namespace
-	// we add it to the map and increment active instance count by phase
-	// as well as total active instance count
-	if _, ok := ActiveInstanceMap[request.Namespace]; !ok {
-		if newPhase != "" {
-			ActiveInstanceMap[request.Namespace] = newPhase
-			ActiveInstancesByPhase.WithLabelValues(newPhase).Inc()
-			ActiveInstancesTotal.Inc()
-		}
-	} else {
-		// If we discover an existing instance's phase has changed since we last saw it
-		// increment instance count with new phase and decrement instance count with old phase
-		// update the phase in corresponding map entry
-		// total instance count remains the same
-		if oldPhase := ActiveInstanceMap[argocd.Namespace]; oldPhase != newPhase {
-			ActiveInstanceMap[argocd.Namespace] = newPhase
-			ActiveInstancesByPhase.WithLabelValues(newPhase).Inc()
-			ActiveInstancesByPhase.WithLabelValues(oldPhase).Dec()
+	// Fetch the value of the $LABEL_SELECTOR environment variable from the argocd instance
+	var labelSelectorValue string
+	for _, envVar := range argocd.Spec.Server.Env {
```

Comment:
The LABEL_SELECTOR value will not be in `argocd.Spec.Server.Env` instead we would need to find the label in `argocd.metadata.labels` and search for the specified key in label selectors.